internal/receiver/discoveryreceiver/endpoint_tracker.go

```diff
@@ -41,6 +45,9 @@ var identifyingAttrKeys = []string{
 	sourcePortAttr,
 }
 
+// k8sPodRegexp is a regular expression to extract the owner object name from a pod name.
+var k8sPodRegexp = regexp.MustCompile(`^(.+?)-(?:(?:[0-9bcdf]+-)?[bcdfghjklmnpqrstvwxz2456789]{5}|[0-9]+)$`)
```

Comment:
This is a very specific regex, where does it come from? It seems worth to mention it in the comment.

Reply:
I built it myself some time ago for the same purpose in the helm chart https://github.com/signalfx/splunk-otel-collector-chart/blob/main/helm-charts/splunk-otel-collector/templates/config/_common.tpl#L326

This will go away once [this todo](https://github.com/signalfx/splunk-otel-collector/pull/4760/files#diff-a9c748957a53e64cfdb2339516836bafc0672ede85d9ee9cb47be3a4c4ebaafbR385-R386) is resolved

Reply:
added more details to the comment

Reply:
:) comment from the link: "This reduces the chances of bad words and
 ensures that strings generated from hash functions appear consistent throughout the API."